File: InvestmentTrackerAPI/InvestmentTrackerAPI/Models/Transaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace InvestmentTrackerAPI.Models;

public class Transaction
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    public DateTime Date { get; set; }

    public decimal Amount { get; set; }
    public int CategoryId { get; set; }

    public string? Description { get; set; }

    public int TypeId { get; set; }

    public virtual Category Category { get; set; } = null!;

    public virtual Type Type { get; set; } = null!;
}

File: InvestmentTrackerUI/shared/types/Transaction.ts
import type { Category } from "./Category"
import type { Type } from "./Type"

export class Transaction {
    id? : string
    typeId? : number
    categoryId? : number
    amount: number
    description? : string
    category? : Category
    type? : Type
    date? : Date

    constructor(data: Partial<Transaction>) {
    this.id = data.id
    this.typeId = data.typeId
    this.categoryId = data.categoryId
    this.amount = data.amount ?? 0
    this.description = data.description
    this.category = data.category
    this.type = data.type
    this.date = data.date
  }

}